helpers/secrets.go

```diff
@@ -651,3 +699,37 @@ func makeK8sData[V any](secretData map[string]V, extraData map[string][]byte,
 func NewSecretsDataBuilder() *SecretDataBuilder {
 	return &SecretDataBuilder{}
 }
+
+// MakeHVSShadowSecretData converts a list of HVS OpenSecrets to k8s secret data.
+func MakeHVSShadowSecretData(secrets []*models.Secrets20231128OpenSecret) (map[string][]byte, error) {
+	data := make(map[string][]byte)
+	for _, v := range secrets {
+		if v.DynamicInstance == nil {
+			continue
+		}
+		secretData, err := marshalJSON(v)
```

Comment:
Would it make sense to use `v.MarshalBinary()` instead?

Reply:
Sure, that works, changed in 3ff3d13